src/ConnectedMode.UnitTests/Suppressions/ServerIssuesStoreTests.cs

```diff
@@ -30,6 +34,88 @@ public class ServerIssuesStoreTests
         public void MefCtor_CheckIsExported()
         {
             MefTestHelpers.CheckTypeCanBeImported<ServerIssuesStore, IServerIssuesStore>();
+            MefTestHelpers.CheckTypeCanBeImported<ServerIssuesStore, IServerIssuesStoreWriter>();
+        }
+
+        [TestMethod]
+        public void AddIssues_ListOfIssuesAsExpected()
```

Comment:
I would probably split these into separate tests:
- dat-driven test: add to empty list, replace=true/false
- add to non-empty list, replace=true -> old issues replaced etc.
- add to non-empty list, replace=false -> old issues retained

I think you've covered all these cases, but it's a bit hard to tell with them all in the same test.

Reply:
You could add an "event raised" check to all of these tests too, then you wouldn't need the separate `AddOrUpdateIssues_EventIsInvoked` test below.